Use StrictMode VmPolicy only in debug build type

adamszewe opened the pull request:
## Description 
`StrictMode` makes sense mainly in for debug build types, enabling it for release does not provide any value as it can add additional overhead and its logs going to be removed (because R8 strips them).





## Changes
* Enable the StrictMode VM policy only for debug build type

imsodin commented:
I am not not that convinced by the benefits of of the content provider. Aren't we abusing a it, as it's intended to do something else? And those other checks you mentioned, couldn't we do that in `Application` as well?

adamszewe commented:
> I am not not that convinced by the benefits of of the content provider. Aren't we abusing a it, as it's intended to do something else? And those other checks you mentioned, couldn't we do that in `Application` as well?

Hello 👋 
Using a `ContentProvider` is a "trick" other libraries, like Firebase uses among many, use.
I think I over-engineered a bit the solution initially thus I rewrote the PR and just moved the code that sets the VMPolicy to a dedicated function which is invoked only when the build type is `debug`.
